Transport saga concurrency behavior documentation (#4528)
Particular/docs.particular.net: persistence/mongodb/index.md

When the outbox is enabled, the deduplication data is kept for seven days by default. To customize this time frame, use the following API:

snippet: MongoDBOutboxCleanup

## Saga concurrency behavior

> MongoDB applies [optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) when saga instance are updated concurrently. When this happens the following error will be visible in the log:

Author: ```suggestion
MongoDB applies [optimistic concurrency control](https://en.wikipedia.org/wiki/Optimistic_concurrency_control) when saga instance are updated. It does not support locking. If optimistic concurrency control conflicts occur the message needs to be reprocessed via [recoverability](/nservicebus/recoverability/).
```

> 

Author: ```suggestion
Please read the guidance on [saga concurrency](/nservicebus/sagas/concurrency.md) on potential improvements.
```

> Creating instance race condition error:

Author: ```suggestion
### Concurrent access to non-existing saga instances
```

```
MongoDB.Driver.MongoCommandException: Command insert failed: WriteConflict.
```

> Updating instances race condition error:

Author: ```suggestion
### Concurrent access to existing saga instances
```

> Please read the guidance on [saga concurrency](/nservicebus/sagas/concurrency.md) on potential improvements.

Author: ```suggestion
```

```
MongoDB.Driver.MongoCommandException: Command update failed: WriteConflict.
```

> Please read the guidance on [saga concurrency](/nservicebus/sagas/concurrency.md) on potential improvements.

Author: ```suggestion
```
